src/main/java/edu/cornell/kfs/sys/document/service/impl/CUFinancialSystemDocumentServiceImpl.java

```diff
@@ -267,9 +268,8 @@ protected AdHocRoutePerson buildFyiRecipient(String userId) {
     }
     
     protected FinancialSystemTransactionalDocumentAuthorizerBase getDocumentAuthorizer(Document doc) {
-    	DataDictionaryService dataDictionaryService = SpringContext.getBean(DataDictionaryService.class);
-        final String docTypeName = dataDictionaryService.getDocumentTypeNameByClass(doc.getClass());
-        Class<? extends DocumentAuthorizer> documentAuthorizerClass = (Class<? extends DocumentAuthorizer>) dataDictionaryService.getDataDictionary().getDocumentEntry(docTypeName).getDocumentAuthorizerClass();
+        DocumentDictionaryService documentDictionaryService = SpringContext.getBean(DocumentDictionaryService.class);
+        Class<? extends DocumentAuthorizer> documentAuthorizerClass = (Class<? extends DocumentAuthorizer>) documentDictionaryService.getDocumentEntry(doc.getClass().getName()).getDocumentAuthorizerClass();
```

Comment:
Is this the correct service method for retrieving the document entry based on document class (or classname)? This might need to be changed to either call the getDocumentEntry() method that accepts a document instance, or to call the getDocumentEntryByClass() method.

Reply:
updated